### PR TITLE
Implemented a POC for "functional" logging

### PR DESCRIPTION
**Pull request by leobcgov**
# Fixes or Implements [AB#8850](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/8850)

* [X] Enhancement
* [ ] Bug
* [ ] Documentation

## Reviewer Guidelines

A reviewer should examine the PR if the Status above indicates the PR is **Ready**.
Once ready, the reviewer should read the information below as well as supporting information like the Azure ticket to ensure they have a good understanding of the request.

The reviewer should then:

1. Is the PR template complete and accurate?

    Confirm by reading all sections below.

1. Confirm the merge branch

    A normal feature/bug branch should merge into the Dev branch.  Exceptions for hotfix and sprint merges do occur and the target would then be Master.

1. Do the code commits have an appropriate comment and include linkage to the azure ticket by including a reference to "AB#nnnn"? 

1. Review the Code Changes

    1. Do the changes match the description?

        The code commits should reflect the documented description and nothing more.  If additional changes are included the reviewer should call these out and request that the developer document the changes in Azure and in the description or remove the changed code.

    1. useful tips

        * Any obvious errors or omissions?
        * Does the code match the style of other similar work?
        * Is the code easy to understand?
        * Is the code well documented?
        * DB Migrations are appropriate?

If the Answer to any of the questions above is No, then a request for change should be the result of the review.

## Description
- Introduced a new way, aka "functional way", for logging on client side.
- Please note that the error will be logged and rethrown (which is missing from the screenshot below).
- The "functional way" of logging is a principle of the Functional Programming which is recommended as Best Practices set for Clean Code.
image

## Testing

* [ ] Unit Tests Updated
* [ ] Functional Tests Updated
* [X] Not Required

### Steps to Reproduce:

If this is a bug, please provide details on how to reproduce the code.

### UI Changes
NO

### Browsers Tested

* [X] Chrome
* [ ] Safari
* [ ] Edge
* [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant.  Include any specialized deployment steps here.

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/src/services/winstonLogger.ts [hunk 1]
```diff
@@ -6,7 +6,27 @@ import { ExternalConfiguration } from "@/models/configData";
...
+    public addLogging: any = (callerName: string, fn: any) => (
+        ...args: any[]
+    ) => {
+        this.logger.debug(
+            `entering ${callerName}.${fn.name}: input args = ${args}`
+        );
+        try {
+            const valueToReturn = fn(...args);
+            this.logger.debug(
+                `exiting ${callerName}.${fn.name}: valueToReturn = ${
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/src/services/winstonLogger.ts:18]: this looks too close to tracing, I think we don't need it right now

File: Apps/WebClient/src/ClientApp/src/components/timeline/linearTimeline.vue [hunk 2]
```diff
@@ -264,5 +268,18 @@ export default class LinearTimelineComponent extends Vue {
         );
         this.currentPage = Math.floor(index / this.numberOfEntriesPerPage) + 1;
     }
+
+    private addLogging(): void {
+        const logger: ILogger = container.get(SERVICE_IDENTIFIER.Logger);
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/src/components/timeline/linearTimeline.vue:273]: this is an interesting example. We can check this out later if we want a full trace of the methods. For now I think the basic logger.debug where necessary should suffice

File: Apps/WebClient/src/ClientApp/src/services/winstonLogger.ts (hunk 1 above)
Comment (leobcgov) [reply Apps/WebClient/src/ClientApp/src/services/winstonLogger.ts:18]: right, which is why it is logged at the debug level. Btw, the exception handling block is interesting. On the side note, turning on this code helps to understand the performance of our code as I'm seeing the getHeadingDate method is being call over 10 time. Without this verbose/debug logging code, the performance issue like this one is hidden.

File: Apps/WebClient/src/ClientApp/src/components/timeline/linearTimeline.vue (hunk 2 above)
Comment (leobcgov) [reply Apps/WebClient/src/ClientApp/src/components/timeline/linearTimeline.vue:273]: exactly, we don't have to go all-in into this for now. The question may be if we want to merge this code to dev or just keep it as an example on this branch only.

**Comment by leobcgov** [comment]
We have decided to not merging this to dev for now. We will revisit this in the future. Please don't delete this PR.

**Pull request closed without merging**